python/tk_multi_publish2/base_hooks/publish_plugin.py

```diff
@@ -451,15 +451,18 @@ def create_settings_widget(self, parent):
 
     def get_ui_settings(self, widget):
         """
-        Invoked by the publisher when the selection changes so the new settings
-        can be applied on the previously selected tasks.
+        Invoked by the publisher when the selection changes. It is performed on the
+        task that was just previously selected so that the publisher can gather the settings.
+        These settings can then be later used to repopulate the custom UI if the task
+        gets selected again, and also be used in the publishing process by the plugin.
 
         The widget argument is the widget that was previously created by
         `create_settings_widget`.
 
         The method returns an dictionary, where the key is the name of a
         setting that should be updated and the value is the new value of that
```

Comment:
I'm still confused here. It says the value in the returned dictionary is the "new value", but we're not actually modifying anything, and we're trying to get *old* values, no?

Reply:
I read this as we have some settings, and we have the UI. When we unselect the plugin, we grab the state and update the settings with new values. We may not update all settings everytime, so some values may stay. I think this is how it works, @jfboismenu  can you confirm?

Reply:
Yup